packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -61,78 +77,171 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
+                this.menuWidth = rect.width;
+                this.forceUpdate();
+            }, 0);
+        }
     }
 
     componentWillUnmount() {
-        document.removeEventListener("mouseup", this._handleInteract);
-        document.removeEventListener("touchend", this._handleInteract);
+        this.removeEventListeners();
+    }
+
+    updateEventListeners() {
+        if (this.props.open) {
+            this.addEventListeners();
+        } else {
+            this.removeEventListeners();
+        }
+    }
+
+    addEventListeners() {
+        // This is click instead of mouse up because events happen in this order
+        // mousedown -> mouseup -> click for mouse presses. We want the event
+        // to be handled by the onClick callback on the menu first before
+        // attempting to handle this event (which would close the menu). The
+        // menu callback stops propagation of the event.
+        document.addEventListener("click", this._handleInteract);
+        document.addEventListener("touch", this._handleInteract);
+        // This is still keyup because we are listening for the escape key,
+        // which invokes events keydown and keyup and NOT keypress.
+        document.addEventListener("keyup", this._handleKeyup);
+    }
+
+    removeEventListeners() {
+        document.removeEventListener("click", this._handleInteract);
+        document.removeEventListener("touch", this._handleInteract);
         document.removeEventListener("keyup", this._handleKeyup);
     }
 
     _handleInteract = (event: {target: any}) => {
-        if (this.node && this.node.contains(event.target)) {
-            return;
-        }
         if (this.props.open) {
-            this.props.onClose();
+            this.props.onDropdownChanged(false);
         }
     };
 
     _handleKeyup = (event: KeyboardEvent) => {
         if (this.props.open && event.key === "Escape") {
             event.preventDefault();
             event.stopImmediatePropagation();
-            this.props.onClose();
+            this.props.onDropdownChanged(false);
+        }
+    };
+
+    updatePosition = () => {
+        const {openerNode} = this.props;
+        if (openerNode) {
+            //$FlowFixMe typing is weird for the return value of ref callback
+            const rect = openerNode.getBoundingClientRect();
+            this.openerBounds = {
+                top: rect.top + window.pageYOffset,
+                left: rect.left + window.pageXOffset,
+                right: rect.right,
+                width: rect.width,
+            };
         }
     };
 
+    findTranslatedPosition() {
+        if (this.openerBounds) {
+            const top = this.openerBounds.top;
+            const left = this.openerBounds.left;
+            const width = this.openerBounds.width;
+            if (this.props.alignment === "left") {
+                return {
+                    transform: `translate3d(${left}px, ${top}px, 0)`,
```

Comment:
@somewhatabstract Should we be using Popper.js to handle the positioning on something like this?